src/spikeinterface/preprocessing/zero_channel_pad.py

```diff
@@ -119,11 +126,12 @@ def get_original_traces_shifted(self, start_frame, end_frame, channel_indices):
         """
         original_start_frame = max(start_frame - self.padding_start, 0)
```

Comment:
```suggestion
        original_start_frame = min(max(start_frame - self.padding_start, 0) self.num_samples_in_original_segment), 
```

Reply:
As the `original_end_frame` is capped at `self.num_samples_in_original_segment`, the same cap should be applied to `original_start_frame`. 

Another alternative, is to remove the two caps.

Reply:
See here:

https://github.com/SpikeInterface/spikeinterface/pull/1984

Where the tests are passed after this modification.

Reply:
I think this looks good! I believe there are 7 cases to handle here (which are captured already in your tests), and involve combinations of  data with (padding start only, padding end only, padding start and end) x (index start padding + data, index data only, index end padding + data). I believe this code handles all cases well. An extra benefit of handling the padding-only case at the level above would be they do not need to be considered here.